### PR TITLE
Add function to calculate Pausch

### PR DESCRIPTION
**Pull request by JulianRoesner**
# Short Description
We want to be able to calculate the Pauschbetrag (see [this ticket](https://steuerlotse.atlassian.net/browse/STL-1678)). I have added a function to calculate the Pauschbetrag following [this table](https://docs.google.com/spreadsheets/d/1UeZCD3Aa8SOZfEs6cZLYq1kLxGL3EKfBiavFCLgfrBc/edit?usp=sharing).
I decided to place it into pauschalbetrag.py because I think this method will probably only be used by the corresponding step.

# Changes
- Add calculation function and tests

# Feedback
- Does the placement make sense to you?
- Do you think we should omit the not needed merkzeichen in the function declaration? I have added them to clarify what we need but they do not make any difference for the result
- Are any tests missing? I tried to test all the edge cases for the different combinations.

# How to review this Pull Request
[Link to Confluence](https://digitalservice4germany.atlassian.net/wiki/spaces/STL/pages/117637157/Development+Process#Approve-or-comment) (internal)

### CODE REVIEW

File: webapp/app/forms/steps/lotse/pauschbetrag.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
+def calculate_pauschbetrag(pflegegrad=False, beh_grad=None, merkzeichen_bl=False, merkzeichen_tbl=False, merkzeichen_h=False,
+                           merkzeichen_ag=False, merkzeichen_g=False):
```
Comment (frederike-ramin) [inline webapp/app/forms/steps/lotse/pauschbetrag.py:2]: I think we should talk about naming of the fields across our different PRs. (e.g. I used english names^^)

Comment (frederike-ramin) [inline webapp/app/forms/steps/lotse/pauschbetrag.py:2]: > Do you think we should omit the not needed merkzeichen in the function declaration? I have added them to clarify what we need but they do not make any difference for the result

I completely get why you included them and I think I'd actually prefer them being there..*but* at some point we're gonna delete them (if not done automatically) and also: if they're not needed, we'd need to test that setting them doesn't change anything and that creates unnecessary test cases that we can avoid by just not being able to set them here.

File: webapp/tests/forms/steps/lotse/test_pauschbetrag.py [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+            'merkzeichen_h': False,
+            'merkzeichen_ag': False,
+            'merkzeichen_g': False,
+        }
+
+        for beh_grad, expected_result in input_output_pairs:
+
+            calculated_pauschbetrag = calculate_pauschbetrag(**params, beh_grad=beh_grad)
+
+            assert calculated_pauschbetrag == expected_result
```
Comment (frederike-ramin) [inline webapp/tests/forms/steps/lotse/test_pauschbetrag.py:32]: So nice taking the params!
(I just dislike the empty lines 29 and 31 - thought I'd mention it if you maaayyybe didn't add them on purpose. 🙈 )

**Review by frederike-ramin — APPROVED** [review]
Very smooth method!

> Does the placement make sense to you?

So that I don't understand you wrong: what else would you place in that file? the calculation of fahrkosten pauschbetrag?

> Are any tests missing? I tried to test all the edge cases for the different combinations.

No way there are tests missing. This is so well tested and the method is also really nice. I'm impressed. 👍

File: webapp/app/forms/steps/lotse/pauschbetrag.py (hunk 1 above)
Comment (JulianRoesner) [reply webapp/app/forms/steps/lotse/pauschbetrag.py:2]: Yes, I assumed that we have to change those.

File: webapp/app/forms/steps/lotse/pauschbetrag.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    if pflegegrad:
+        return 7400
+    if merkzeichen_bl:
+        return 7400
+    if merkzeichen_tbl:
+        return 7400
+    if merkzeichen_h:
+        return 7400
```
[the remark this answers is not in the document]
Comment (JulianRoesner) [reply webapp/app/forms/steps/lotse/pauschbetrag.py:10]: I like an OR more -> 👍🏼

**Comment by JulianRoesner** [comment]
> So that I don't understand you wrong: what else would you place in that file? the calculation of fahrkosten pauschbetrag?

No, I would have thought that we put the StepPauschbetrag here.

**Comment by frederike-ramin** [comment]
> > So that I don't understand you wrong: what else would you place in that file? the calculation of fahrkosten pauschbetrag?
>
> No, I would have thought that we put the StepPauschbetrag here.

Ah yes, of course, it is in the step folder afterall. So yes, completely agree with the placement. 🙈

File: webapp/app/forms/steps/lotse/pauschbetrag.py [hunk 4]
```diff
@@ -1,24 +1,34 @@
...
-        40: 860,
-        50: 1140,
-        60: 1440,
-        70: 1780,
-        80: 2120,
-        90: 2460,
-        100: 2840,
-    }
+    if disability_degree > 100 or disability_degree < 20:
+        raise ValueError(f"Disability degree has incorrect value of {disability_degree}. Accept only values between 20 and 100.")
```
Comment (frederike-ramin) [inline webapp/app/forms/steps/lotse/pauschbetrag.py:15]: I'm not sure this and the corresponding tests are necessary or even desired. Correct me if I'm wrong, but no incorrect value could *possibly* get here, right? Because we first validate that the input is correct and only expect valid inputs here? Having this here makes me question that.

File: webapp/tests/forms/steps/lotse/test_pauschbetrag.py [hunk 5]
```diff
@@ -1,37 +1,52 @@
...
             (50, 1140),
+            (55, 1140),
             (60, 1440),
+            (65, 1440),
             (70, 1780),
+            (75, 1780),
             (80, 2120),
+            (85, 2120),
             (90, 2460),
+            (95, 2460),
```
Comment (frederike-ramin) [inline webapp/tests/forms/steps/lotse/test_pauschbetrag.py:25]: This on the other hand is a great catch that we definitely need to test and make sure works correctly.

File: webapp/tests/forms/steps/lotse/test_pauschbetrag.py [hunk 6]
```diff
@@ -34,9 +34,7 @@ def test_if_no_merkzeichen_or_pflegegrad_set_then_return_correct_value_for_disab
         }
 
         for disability_degree, expected_result in input_output_pairs:
-
             calculated_pauschbetrag = calculate_pauschbetrag(**params, disability_degree=disability_degree)
-
```
Comment (frederike-ramin) [inline webapp/tests/forms/steps/lotse/test_pauschbetrag.py]: Thanks! 🙏

**Review by frederike-ramin — COMMENTED** [review]
I think we could talk about the < 20 > 100 thing. But the 5-er-steps (dat denglisch^^): great catch. :)

File: webapp/app/forms/steps/lotse/pauschbetrag.py (hunk 4 above)
Comment (JulianRoesner) [reply webapp/app/forms/steps/lotse/pauschbetrag.py:15]: I hoped you would say this and I would be fine to add that to the documentation and get rid of those lines and the corresponding tests.

Comment (frederike-ramin) [reply webapp/app/forms/steps/lotse/pauschbetrag.py]: Haha, pokering with PRs - I like it. 😂

**Review by frederike-ramin — APPROVED** [review]
